lang/java/reef-checkpoint/src/main/java/org/apache/reef/io/checkpoint/fs/FSCheckpointService.java

```diff
@@ -75,78 +75,90 @@ static final Path tmpfile(final Path p) {
   public CheckpointWriteChannel create()
       throws IOException {
 
-    final String name = namingPolicy.getNewName();
-
+    final String name = namingPolicy.getNewCheckpointName();
     final Path p = new Path(name);
     if (p.isUriPathAbsolute()) {
-      throw new IOException("Checkpoint cannot be an absolute path");
+      throw new IOException("Checkpoint name cannot be an absolute path.");
     }
-    return createInternal(new Path(base, p));
+
+    return createInternal(new Path(basePath, p));
   }
 
   CheckpointWriteChannel createInternal(final Path name) throws IOException {
-
     //create a temp file, fail if file exists
```

Comment:
> it is to check whether the file is indeed writable

@markusweimer  in that phrase, did you refer to the temporary file, final destination file or to the fact that a checkpointing process may not have write access to a file system ?